libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -197,9 +204,27 @@ - (void)send:(SFRestRequest *)request delegate:(id<SFRestDelegate>)delegate shou
         }];
     } else {
 
-        // Auth credentials exist. Just send the request.
-        [self enqueueRequest:request delegate:delegate shouldRetry:shouldRetry];
+        /*
+         * Auth credentials exist. Send the request if an OAuth session is not being refreshed.
+         * Otherwise, wait for the current session refresh call to complete before sending.
+         */
+        if (!self.sessionRefreshInProgress) {
+            [self enqueueRequest:request delegate:delegate shouldRetry:shouldRetry];
+        }
+    }
+}
+
+- (SFOAuthSessionRefresher *)sessionRefresherForUser:(SFUserAccount *)user {
+    @synchronized (self) {
+        self.sessionRefreshInProgress = YES;
```

Comment:
Fixed this in my latest commit [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2152/commits/f70d0b4bc685edfdd12e8f2cd85a1ec910c1e199).